### DISCUSSION TITLE
How enso build his GUI i can see you did not use javascript

### DISCUSSION THREAD

**Discussion by khanakia** [enso-org/ide#1622] [General]
I am curious to know How enso built his GUI i can see you did not use javascript so what other libraries are using to build the GUI ?

Is it OpenGL, Webgl or what ?

**Comment by MichaelMauderer** [comment]
Hi @khanakia! We are building the IDE in Rust, compiled to WASM. We've got our own rendering library EnsoGL built on top of WebGL.

**Reply by khanakia** [reply]
Thank you :)

**Comment by wdanilo** [comment]
Hi @khanakia !
We are using a custom WebGL-based renderer. It's written in Rust and compiled to WASM. You can learn more about it here: https://github.com/enso-org/ide/tree/develop/src/rust/ensogl :)

**Reply by khanakia** [reply]
Thank you :)

**Comment by wdanilo** [comment]
@MichaelMauderer damn, you were 15 seconds faster :D
